dsn.go

```diff
@@ -45,6 +45,7 @@ type Config struct {
 	AllowAllFiles           bool // Allow all files to be used with LOAD DATA LOCAL INFILE
 	AllowCleartextPasswords bool // Allows the cleartext client side plugin
 	AllowOldPasswords       bool // Allows the old insecure password method
+	AllowNativePasswords	bool // Allows mysql native password authentication method
```

Comment:
remove "mysql". It's obvious were not using MongoDB's native password authentication method